test/unit/ajax.js

```diff
@@ -1590,6 +1592,19 @@ module( "ajax", {
 				ok( false, "error" );
 			}
 		} );
+
+		if ( jQuery.support.cors ) {
+			ajaxTest( "gh-1684 - jQuery.ajax() - support cross-domain PATCH in IE10-11", 1, {
+				url: "http://httpbin.org/patch",
```

Comment:
> How would you test jQuery.ajaxSettings.xhr?

In IE env set `type` to `patch` and call `jQuery.ajaxSettings.xhr` then test returned value.

> You cannot test if a native API bugs is worked around properly if you replace those native APIs with stubs that won't reproduce these bugs.

In technical terms, testing native API with conjunction of `ajax` module would be an integration test, testing through stubbing interface would be a unit test.

As an option, we can settle on unit test now and create issue for integration one and after we improve testing infrastructure introduce the intergration test.